mongo/options/clientoptions.go

```diff
@@ -805,6 +811,16 @@ func (c *ClientOptions) SetRetryReads(b bool) *ClientOptions {
 	return c
 }
 
+// SetSecurityToken sets a security token that is sent with every command.
+//
+// Deprecated: This option is for internal use only and should not be set. It
+// may be changed or removed in any release.
+func (c *ClientOptions) SetSecurityToken(token string) *ClientOptions {
```

Comment:
Instead of adding a deprecated method to the stable API, it's possible to do this as a deployment setting. There are two pros to this solution:

1. It makes doing this intentionally difficult, a la reflection
2. It avoid touching most of the stable API

Here is the intended use case:

```go

func main() {
	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017")

	topoCfg, err := topology.NewConfig(clientOpts, nil)
	if err != nil {
		panic(err)
	}

	topoCfg.SecurityToken = "x"

	topo, err := topology.New(topoCfg)
	if err != nil {
		panic(err)
	}

	stopo := &topology.SecurityTokenTopology{
		Topology: topo,
	}

	clientOpts.Deployment = stopo

	client, err := mongo.Connect(context.Background(), clientOpts)
	if err != nil {
		panic(err)
	}

	// cont...
}
```

Here is a commit with the implementation, for your review / local testing: https://github.com/prestonvasquez/mongo-go-driver/commit/ec037850b721fc01bf42f4c83176f0044c49c61b